finances/views.py

```diff
@@ -1843,30 +1839,41 @@ def form_valid(self, form, *args, **kwargs):
             isParticipant = False
 
         errors = []
+        nb_empty_rows = 0
         i = 1
 
         # Enregistrement des pondérations
         for row in rows:
             i += 1;
             try:
-                username = row[0].value.strip() # Should be a str
-                user = User.objects.get(username=username)
-                try:
-                    pond = int(row[1].value) # Should be an int. Else, raise an error
-
-                    if pond > 0:
-                        self.se.change_weight( user, pond, isParticipant )
-                except:
-                    errors.append( "Erreur avec " + username + " (ligne n*" + str(i) + "). A priori pas ajouté." )
+                if row[0].value and row[1].value:
+                  username = row[0].value.strip() # Should be a str
+                  user = User.objects.filter(username=username)
+                  if user.count() > 0:
+                      user = User.objects.get(username=username)
+                      try:
+                          pond = int(row[1].value) # Should be an int. Else, raise an error
+                          if pond > 0:
+                              self.se.change_weight( user, pond, isParticipant )
+                      except:
+                          errors.append( "Erreur avec " + username + " (ligne n*" + str(i) + "). A priori pas ajouté." )
+                  else:
+                      errors.append( "L'utilisateur " + username + " n'existe pas. (ligne n*" + str(i) + ").")
```

Comment:
If user = User.objects.filter(username=username) doesn't find a user, i'm pretty sure it raise an error (catched by line 1865). Are you sure it's relevant ?

Reply:
Nope, there is no error, when user is not found, the query result will be empty.
However, there is no need to store first that query into a variable.

Reply:
Oh yeah, you are using filter and not get... Why that, btw ?

Reply:
To check if a user exists before trying to add it, get generates an error, but not filter (as long as i'm using existing object attributes ...), and show a specific error when user is not found 😉